integrations/access/jira/app.go

```diff
@@ -299,7 +311,32 @@ func (a *App) onJiraWebhook(ctx context.Context, webhook Webhook) error {
 		"jira_issue_key": issue.Key,
 	})
 
+	webHookStatusName := strings.ToLower(webhook.Issue.Fields.Status.Name)
 	statusName := strings.ToLower(issue.Fields.Status.Name)
+
+	// Retry until API syncs up with webhook payload.
+	if webHookStatusName != statusName {
+		retry, err := retryutils.NewLinear(a.retryConfig)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		err = retry.For(ctx, func() error {
+			issue, err = a.jira.GetIssue(ctx, webhook.Issue.ID)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+			statusName = strings.ToLower(issue.Fields.Status.Name)
+			if webHookStatusName != statusName {
+				return trace.CompareFailed("mismatch of webhook payload and issue API response: %q and %q",
+					webHookStatusName, statusName)
+			}
+			return nil
+		})
+		if err != nil {
+			log.Warnf("Using most recent successful getIssue response: %v", err)
+		}
```

Comment:
I think we could move the first call to the retryer, this way there was only one place where we call the API.